Source/WebCore/Modules/compression/CompressionStream.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#pragma once
+
+#include <wtf/IsoMalloc.h>
+#include <wtf/IsoMallocInlines.h>
+#include <wtf/RefCounted.h>
+#include <wtf/text/WTFString.h>
+#include "ActiveDOMObject.h"
+#include "Document.h"
+#include "Event.h"
+#include "EventTarget.h"
+#include "ExceptionOr.h"
+#include "WritableStream.h"
+#include "ReadableStream.h"
+#include "ReadableStreamSource.h"
+#include "ScriptExecutionContext.h"
+#include <iostream>
+
+namespace JSC {
+class CallFrame;
+class JSGlobalObject;
+class JSValue;
+}
+
+namespace WebCore {
+
+class CompressionStream : public RefCounted<CompressionStream>,
+    public ActiveDOMObject,
+    public EventTargetWithInlineData {
+    WTF_MAKE_ISO_ALLOCATED(CompressionStream);
+
+public:
+    // Compression Formats
+    // gzip         "GZIP file format"              [RFC1952]
+    // deflate      "ZLIB Compressed Data Format"   [RFC1950]
+    // deflate-raw  "The DEFLATE algorithm"         [RFC1951]
+    enum CompressionFormat { gzip, deflate, deflate_raw };
+
+    CompressionStream(Document& document, String string, CompressionFormat format) : ActiveDOMObject(document), m_string(string), m_format(format) {
+        
```

Comment:
Nit: I don't think we need the blank lines around the small implementation here.